### ISSUE TITLE
Code compilation error and white screen

### ISSUE THREAD

**Issue by RobertMorton56** [mean00/DSO_STM32Duino#6]
Hi,

Firstly, thank you for creating this code. It's brought my DSO back to life, a huge improvement over the stock firmware.

I managed to compile the git code to a bin, but there's an error in /src/dso_debug.h as far as I can tell. There's a trailing "²" on line 7. The code then compiles but when flashed the device just has a white screen. I'm compiling with "cmake -DMCU_SPEED=128000000 -DUSE_VANILLA_HW=True". 

Can you help?

**Comment by mean00** [comment]
Thanks for the notification about the extra 2.
The crash at start is usually one of two things : 
- Problem with the screen
- Problem flashing the code and/or code too big

How did you flash your dso ?
I'm rebuilding latest git master locally, i may have broken something due to usb support

**Comment by mean00** [comment]
rebuild locally from master, not tested yet
[Dso150STM32F103_vanilla_72M.elf.gz](https://github.com/mean00/DSO_STM32Duino/files/6398139/Dso150STM32F103_vanilla_72M.elf.gz)

**Comment by RobertMorton56** [comment]
Thanks for the reply. I'm using Flash Loader Demonstrator (from STMicroelectronics) - it was the path of least resistance for me. I used Linux to compile but all the tools to flash the images weren't playing ball. And as it would happen Flash Loader Demonstrator doesn't support elf files AFAIK so I can't test your rebuild.

My screen is working fine, I flashed release 1.06 and I'm back up and running. I was looking for serial dump support hence me trying a master build.

**Comment by mean00** [comment]
Could you try flashing the bin extracted from the elf i posted above ?
That will help finding out if it is a generation issue
I.e. it that one works it means something is amiss regarding the code generation (compiler/....) but the code by itself works
I'll try on mine later on

**Comment by RobertMorton56** [comment]
I'm getting the same white screen when I boot the extracted binary.

**Comment by mean00** [comment]
I just tried flashing the elf on mine with a blackmagic probe
It works
Could you power it on and just wait to see if eventually it comes online ?

**Comment by RobertMorton56** [comment]
I left it powered on for about an hour. Still a white screen. It must be the way I'm flashing it?

**Comment by mean00** [comment]
That's weird, it works for you with v1.06 and it's ~ the same

**Comment by RobertMorton56** [comment]
Very strange indeed. I've just reverted to v1.06 and it boots fine. Just managed to use stm32flash to flash your elf binary in Linux, same effect... white screen not booting.

Edit: Managed to upload my own master build with stm32flash! Great. The TLDR is don't use "Flash Loader Demonstrator" to upload the firmware. For posterity i used:

`stm32flash -w Dso150_usb_stm32.bin -g 0x8002000 /dev/ttyUSB0`

Thanks for your help!
